app/src/main/java/samyups/example/trainingss/MainActivity.kt

```diff
@@ -1,11 +1,28 @@
 package samyups.example.trainingss
 
-import android.support.v7.app.AppCompatActivity
+import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
+
+    val dogAdapter = DogAdapter()
```

Comment:
This should be private. You aren't accessing it from outside this class, and in general you want to limit what you are exposing in each class.